middle_end/flambda2/to_cmm/to_cmm.ml

```diff
@@ -1321,24 +1331,30 @@ and make_switch ~tag_discriminant env res e arms =
     | (x, if_x), (_, if_not) ->
       wrap (C.ite (C.eq (C.int x) e) ~then_:if_x ~else_:if_not), res)
   (* General case *)
-  | n ->
+  | arms ->
+    let n = List.length arms in
+    let max_d =
+      List.sort
+        (fun (discr1, _) (discr2, _) -> Targetint_31_63.compare discr2 discr1)
+        arms
+      |> List.hd |> fst
```

Comment:
It's a bit wasteful to sort the list only to get the max discriminant. Why not do a simple fold instead ?